File: 2002B2/CS/WinForms-Inheritance-Vehicle/Vehicle/VehicleSampleBO/SailBoat.cs
using System;

namespace VehicleDemo
{
	/// <summary>
	/// Summary description for SailBoat.
	/// </summary>
	public class SailBoat : Vehicle
	{
		//We need to override the ComputeRange function to add support for sailboats
		//For sailboats we want to triple the couputed range.  We will call the
		//ComputeRange function on MyBase which is the Vehicle class
		public override int ComputeRange()
		{
			//This is a sailboat so we triple the calculated range
			return base.ComputeRange() * 3;
		}

		public SailBoat()
		{
			//
			// TODO: Add constructor logic here
			//
		}
	}
}

File: 2002B2/VB/WinForms-Inheritance-Vehicle/Vehicle/VehicleBO/SailBoat.vb
Namespace VehicleDemo
    Public Class SailBoat
        Inherits Vehicle

        'We need to override the ComputeRange function to add support for sailboats
        'For sailboats we want to triple the couputed range.  We will call the
        'ComputeRange function on MyBase which is the Vehicle class
        Overrides Function ComputeRange() As Integer
            'double the calculated range
            ComputeRange = MyBase.ComputeRange * 3
        End Function
    End Class
End Namespace
